### DISCUSSION TITLE
waybar battery module

### DISCUSSION THREAD

**Discussion by codeskin-r** [Matt-FTW/dotfiles#17] [General]
How to add a a battery module in waybar? Thanks!

**Comment by Matt-FTW** [comment]
Hi,

The structure I have is based on some custom modules that you import on a custom bar, and then you import that bar into the config.jsonc from Waybar. This is to allow some modularity instead of having everything in the same file, that sometimes can be a mess to work with.

Having that in mind, open the  [config.jsonc](https://github.com/Matt-FTW/dotfiles/blob/main/.config/waybar/config.jsonc) inside the config folder of Waybar. You can see some objects where each one [imports a bar](https://github.com/Matt-FTW/dotfiles/blob/53ded3988bae6c5942b1f1b99ab769d262ca306e/.config/waybar/config.jsonc#L3). If you go into the specific one you want to use, you will see [some imports](https://github.com/Matt-FTW/dotfiles/blob/53ded3988bae6c5942b1f1b99ab769d262ca306e/.config/waybar/bars/bar-1.jsonc#L2) of the custom modules we have defined. 

I added a new [battery module](https://github.com/Matt-FTW/dotfiles/blob/main/.config/waybar/modules/battery.jsonc). To import it do the same as the others by specifying the path of the module file and then down in the array of modules, add the battery module definition to one of those (left, center or right).

```json
// Module import
"include": [
     "~/.config/waybar/modules/custom-cava.jsonc",
]

...

// Module definition
"modules-right": [
    "battery",
]


```

Now restart Waybar (`killall waybar; waybar`) and you should have the battery module ready to go.

Sorry if I explained to much but I wanted you to understand everything :)

**Reply by codeskin-r** [reply]
Thank you! I was looking and yes, I managed to add a battery module the way you explained it to me, the problem is that it is not updated "live" but when I click on the module. How would I solve this?

**Reply by Matt-FTW** [reply]
Umm thats weird, it should update the amount indicated in the [interval def](https://github.com/Matt-FTW/dotfiles/blob/53ded3988bae6c5942b1f1b99ab769d262ca306e/.config/waybar/modules/battery.jsonc#L9), which is 5 seconds. 

Sorry but I cannot test it because I dont have any laptop. I'd recommend you check the waybar wiki or see if there's any issue related with it. Other than that, sorry!

Be sure to let me know if you found the issue :)

**Reply by codeskin-r** [reply]
You are a genius! Believe me, you are helping me a lot, thank you for adding the battery module, by the way, I take this opportunity to tell you that your configuration seems the best to me! Could you help me add a screen brightness module to the waybar? I am using brightnessctl but I am missing the module

**Reply by Matt-FTW** [reply]
Here's the module: 663e88d 

btw, thanks for your kind words :smile:

**Reply by codeskin-r** [reply]
I need to ask you a few more things, when using workspaces and changing them, why when I go back to them the keyboard shortcuts are no longer there? Is this normal? Or do they have a special behavior?

**Reply by Matt-FTW** [reply]
Sorry but I didnt understand what you ment. Are you referring to the on-click actions? They should be working correctly as far as I know

**Reply by codeskin-r** [reply]
It's a bit difficult to explain :D, I'll try to explain it to you, it happens when I use the "super" key + the numbers, when I go back to the workspace where I opened a certain program, it's like it disappears, I don't know if you understand me.

**Reply by Matt-FTW** [reply]
Okay now i get it. Are you using the hyprsplit plugin? It happened to me when I was using it so maybe thats the issue. 

If not, then I think is an issue with the special workspaces that pyprland uses for the scratchpads. Can you check if the app is still running although you cannot see it?

**Reply by codeskin-r** [reply]
It's an issue with pyperland workspaces, it's obvious that it has to do with scratchpads, but since I hardly use them, I'm not used to it, too much for me but no less useful! LOL, thank you very much for your help, seriously, I can imagine how much work you put into these dotfiles!

**Reply by Matt-FTW** [reply]
Okay nice :). Yeah it was like 1 year of work but it was worth it haha. I'll still keep adding things here and there, but as of know, im just gonna mantain it and make sure is usable in most of the systems.
It's great that i could be of any help, if you have any other doubt or problem, here i am :D
Have a nice weekend!

**Reply by codeskin-r** [reply]
The truth is that yes, it was very worth it, aesthetically it is pleasant, and the animations are going really well, the VS Codium configuration is really good too, and the NVIM configuration uff, no way!

I appreciate all the help, I'll let you know if I need to ask you a few more things.

**Reply by codeskin-r** [reply]
Where should I add the aliases in the fish configuration? Sorry for the newbie question :D

**Reply by Matt-FTW** [reply]
Its fine haha. I'd recommend you to create functions instead of aliases, but you can do aliases too. [Here's an article](https://bsago.me/tech-notes/make-fish-start-faster) that explains it very well. As the article said, be sure to not put them inside the config.fish.

Apart from that, if you wanna **create some functions** out of an alias you can do it like this:

```fish
alias -s <alias-name>='<command>'
```

It will create a function with the alias name and the command inside the functions folder. 

If you want to **create some aliases** instead, I'd recommend to create an aliases.fish file inside the fish dir and to put the aliases there. After that, make sure to import that file in the config.fish like this:

```fish
source ~/.config/fish/aliases.fish
``` 

I have some of that explained [over here](https://github.com/Matt-FTW/dotfiles/tree/main/.config/fish). There are docs for some of the most important configs and apps inside the repo so the next time be sure to check those first.

by the way, for the sake of beign organized, can you create another discussion next time if it doesnt correlate to this one? thanks :)
